dmoj/urls.py

```diff
@@ -51,8 +51,7 @@
         TitledTemplateView.as_view(template_name='registration/registration_closed.html',
                                    title='Registration not allowed'),
         name='registration_disallowed'),
-    url(r'^login/$', auth_views.LoginView.as_view(
-        template_name='registration/login.html',
+    url(r'^login/$', user.CustomLoginView.as_view(
         extra_context={'title': _('Login')},
```

Comment:
I'm pretty sure you can move these directly into `CustomLoginView` so that this part can just call `.as_view()` with no kwargs.